doc/build/dts/bindings-syntax.rst

```diff
@@ -0,0 +1,799 @@
+available. See :ref:`dt-writing-property-values` for more details about writing
+values of each type in a DTS file. See :ref:`dt-phandles` for more information
+about the ``phandle*`` type properties.
+
+.. list-table::
+   :header-rows: 1
+   :widths: 1 3 2
+
+   * - Type
+     - Description
+     - Example in DTS
+
+   * - ``string``
+     - exactly one string
+     - ``status = "disabled";``
+
+   * - ``int``
+     - exactly one 32-bit value (cell)
+     - ``current-speed = <115200>;``
+
+   * - ``boolean``
+     - flags that don't take a value when true, and are absent if false
+     - ``hw-flow-control;``
+
+   * - ``array``
+     - zero or more 32-bit values (cells)
+     - ``offsets = <0x100 0x200 0x300>;``
+
+   * - ``uint8-array``
+     - zero or more bytes, in hex ('bytestring' in the Devicetree specification)
+     - ``local-mac-address = [de ad be ef 12 34];``
+
+   * - ``string-array``
+     - zero or more strings
+     - ``dma-names = "tx", "rx";``
+
+   * - ``phandle``
+     - exactly one phandle
+     - ``interrupt-parent = <&gic>;``
+
+   * - ``phandles``
+     - zero or more phandles
+     - ``pinctrl-0 = <&usart2_tx_pd5 &usart2_rx_pd6>;``
+
+   * - ``phandle-array``
+     - a list of phandles and 32-bit cells (usually specifiers)
+     - ``dmas = <&dma0 2>, <&dma0 3>;``
+
+   * - ``path``
+     - a path to a node as a phandle path reference or path string
+     - ``zephyr,bt-c2h-uart = &uart0;`` or
+       ``foo = "/path/to/some/node";``
+
+   * - ``compound``
+     - a catch-all for more complex types (no macros will be generated)
+     - ``foo = <&label>, [01 02];``
+
+deprecated
+==========
+
+A property with ``deprecated: true`` indicates to both the user and the tooling
+that the property is meant to be phased out.
+
+The tooling will report a warning if the devicetree includes the property that
+is flagged as deprecated. (This warning is upgraded to an error in the
+:ref:`twister_script` for upstream pull requests.)
+
+The default setting is ``deprecated: false``. Using ``deprecated: false`` is
+therefore redundant and strongly discouraged.
+
+.. _dt-bindings-default:
+
+default
+=======
+
+The optional ``default:`` setting gives a value that will be used if the
+property is missing from the devicetree node.
+
+For example, with this binding fragment:
+
+.. code-block:: YAML
+
+   properties:
+     foo:
+       type: int
+       default: 3
+
+If property ``foo`` is missing in a matching node, then the output will be as
+if ``foo = <3>;`` had appeared in the DTS (except YAML data types are used for
+the default value).
+
+Note that combining ``default:`` with ``required: true`` will raise an error.
+
+For rules related to ``default`` in upstream Zephyr bindings, see
+:ref:`dt-bindings-default-rules`.
+
+See :ref:`dt-bindings-example-properties` for examples. Putting ``default:`` on
+any property type besides those used in the examples will raise an error.
```

Comment:
~~Is this enforced? I ask as for a hd44780 driver I am using this for the main binding itself for line addresses, most displays use the same addresses but on the odd chance that there is one somewhere that doesn't, it allows it to be overridden without the majority of displays needing to put common values in here.~~

~~Just had a look through in-tree bindings and it's used in a lot of places (quite a few in displays)~~
After re-reading it a number of times, I see that it means refer to the linked examples to see which types you are allowed to use default on, not that you can only use it in example bindings for sample projects. This text could be clarified a bit better, or list the types allowed here in the text.